goal_src/jak2/engine/debug/history.gc

```diff
@@ -5,5 +5,587 @@
+   (channel-mask uint64   :offset-assert  24)
+   (index        int32    :offset-assert  32)
+   (done?        symbol   :offset-assert  36)
+   )
+  :method-count-assert 12
+  :size-assert         #x28
+  :flag-assert         #xc00000028
+  (:methods
+    (new (symbol type uint32) _type_ 0)
+    (frame-counter-delta (_type_ history-elt) time-frame 9)
+    (update-entries! (_type_) history-elt 10)
+    (history-iterator-method-11 (_type_ history-elt) float 11)
+    )
+  )
+
+(defmethod inspect history-iterator ((obj history-iterator))
+  (when (not obj)
+    (set! obj obj)
+    (goto cfg-4)
+    )
+  (format #t "[~8x] ~A~%" obj (-> obj type))
+  (format #t "~1Tmax-age: ~D~%" (-> obj max-age))
+  (format #t "~1Towner: ~D~%" (-> obj owner))
+  (format #t "~1Tproc: ~A~%" (-> obj proc))
+  (format #t "~1Tout: ~A~%" (-> obj out))
+  (format #t "~1Tchannel-mask: ~D~%" (-> obj channel-mask))
+  (format #t "~1Tindex: ~D~%" (-> obj index))
+  (format #t "~1Tdone?: ~A~%" (-> obj done?))
+  (label cfg-4)
+  obj
+  )
+
+(deftype history (basic)
+  ((alloc-index      int32                :offset-assert   4)
+   (allocated-length int32                :offset-assert   8)
+   (elts             history-elt :dynamic :offset         16)
+   )
+  :method-count-assert 11
+  :size-assert         #x10
+  :flag-assert         #xb00000010
+  (:methods
+    (new (symbol type int) _type_ 0)
+    (history-method-asm-9 () none 9)
+    (clear-history-entries! (_type_) none 10)
+    )
+  )
+
+(defmethod inspect history ((obj history))
+  (when (not obj)
+    (set! obj obj)
+    (goto cfg-4)
+    )
+  (format #t "[~8x] ~A~%" obj (-> obj type))
+  (format #t "~1Talloc-index: ~D~%" (-> obj alloc-index))
+  (format #t "~1Tallocated-length: ~D~%" (-> obj allocated-length))
+  (format #t "~1Telts[0] @ #x~X~%" (-> obj elts))
+  (label cfg-4)
+  obj
+  )
+
+(defmethod length history ((obj history))
+  (-> obj allocated-length)
+  )
+
+(defmethod asize-of history ((obj history))
+  (the-as int (+ (-> obj type size) (* 48 (-> obj allocated-length))))
+  )
+
+(defmethod new history ((allocation symbol) (type-to-make type) (arg0 int))
+  (let ((v0-0 (object-new allocation type-to-make (the-as int (+ (-> type-to-make size) (* 48 arg0))))))
+    (set! (-> v0-0 allocated-length) arg0)
+    v0-0
+    )
+  )
 
+(defmethod clear-history-entries! history ((obj history))
+  "Iterates through each [[history-elt]] in the `elt` dynamic array
+       For each entry:
+       - clear `timestamp`
+       - clear `record-tag`"
+  (set! (-> obj alloc-index) 0)
+  (countdown (v1-0 (-> obj allocated-length))
+    (let ((a1-3 (the-as history-elt (+ (+ (* 48 v1-0) 12) (the-as int obj)))))
+      (set! (-> a1-3 record-tag) (the-as uint 0))
+      (set! (-> a1-3 timestamp) 0)
+      )
+    0
+    )
+  0
+  (none)
+  )
+
+(define-perm *history* history (new 'debug 'history 4096))
+
+(clear-history-entries! *history*)
+
+;; ERROR: function was not converted to expressions. Cannot decompile.
```

Comment:
Looks like this one works if you remove it from `hacks.json`
Maybe
```
(history-method-asm-9 (_type_ int int int) object 9)
```

(with one of those ints being `history-channel`)